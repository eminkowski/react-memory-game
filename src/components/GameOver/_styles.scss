@import '~globals';

.game-over {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100%;
  user-select: none;
  width: 100%;

  &-message {
    animation: pulse 2s infinite linear;
    color: $black;
    font-size: 20px;
    height: 70px;
  }

  &-star {
    animation: spin 2s infinite linear;
    display: inline-block;
  }

  &-text {
    margin: 20px;
  }

  &-button {
    background: none;
    border: 3px solid $black;
    color: $black;
    cursor: pointer;
    display: block;
    font-size: 18px;
    padding: 12px;
    user-select: none;
  }

  &-button:hover {
    background: $beige;
    border-color: $black;
    color: $black;
  }
}

@keyframes pulse {
  0% {
    font-size: 18px;
  }

  50% {
    font-size: 20px;
  }

  100% {
    font-size: 18px;
  }
}

@media (min-width: 500px) {
  @keyframes pulse {
    0% {
      font-size: 24px;
    }

    50% {
      font-size: 26px;
    }

    100% {
      font-size: 24px;
    }
  }
}

@media (min-width: 1000px) {
  @keyframes pulse {
    0% {
      font-size: 32px;
    }

    50% {
      font-size: 34px;
    }

    100% {
      font-size: 32px;
    }
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
