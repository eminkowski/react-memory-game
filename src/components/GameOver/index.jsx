import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './_styles.scss';

export default class GameOver extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    style: PropTypes.string,
    onClick: PropTypes.func.isRequired,
  };

  render() {
    const { className, style, onClick } = this.props;

    return (
      <div className={cn('game-over', className)} style={style}>
        <div className="game-over-message">
          <span className="game-over-star" role="img" aria-label="star">⭐</span>
          <span className="game-over-text">You did it!</span>
          <span className="game-over-star" role="img" aria-label="star">⭐</span>
        </div>
        <button className="game-over-button" onClick={onClick}>
          Play again?
        </button>
      </div>
    );
  }
};
