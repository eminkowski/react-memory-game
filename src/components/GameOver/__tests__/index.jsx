/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import GameOver from '../index';

describe('GameOver', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <GameOver onClick={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render two spans that each contain a star', () => {
    const component = shallow(
      <GameOver onClick={jest.fn()} />
    );
    const star = component.find('.game-over-star');
    expect(star.length).toEqual(2);

    expect(star.get(0).type).toEqual('span');
    expect(star.get(0).props.children).toEqual('⭐');

    expect(star.get(1).type).toEqual('span');
    expect(star.get(1).props.children).toEqual('⭐');
  });

  it('should render a span for the game over text', () => {
    const component = shallow(
      <GameOver onClick={jest.fn()} />
    );
    const text = component.find('.game-over-text');
    expect(text.length).toEqual(1);

    expect(text.get(0).props.children).toEqual('You did it!');
  });

  it('should render a button with the onClick prop', () => {
    const onClick = jest.fn();

    const component = mount(
      <GameOver onClick={onClick} />
    );
    const button = component.find('button');
    expect(button.length).toEqual(1);

    expect(button.get(0).props.onClick).toBe(onClick);
  });
});
