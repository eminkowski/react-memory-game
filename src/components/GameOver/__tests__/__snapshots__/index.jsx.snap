// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver should render correctly 1`] = `
<div
  className="game-over"
  style={undefined}
>
  <div
    className="game-over-message"
  >
    <span
      aria-label="star"
      className="game-over-star"
      role="img"
    >
      ⭐
    </span>
    <span
      className="game-over-text"
    >
      You did it!
    </span>
    <span
      aria-label="star"
      className="game-over-star"
      role="img"
    >
      ⭐
    </span>
  </div>
  <button
    className="game-over-button"
    onClick={[Function]}
  >
    Play again?
  </button>
</div>
`;
